src/Routing/Types.purs

```diff
@@ -0,0 +1,6 @@
+module Routing.Types where
+
+import qualified Data.StrMap as M
+
+data RoutePart = Path String | Query (M.StrMap String)
+type Route = [RoutePart]
```

Comment:
Oh, I'm not suggesting that there only be one query parameter. I just don't see the appeal of distinguishing `foo/?bar=12/baz/?quux=123/` from `foo/?bar=12&quux=123/baz/` as routes. I would normally expect query parameters to be put at the end, and for both of these to fail to parse:

`foo/baz?bar=12&quux=123/`